### ISSUE TITLE
NIP-89 Support Recommended application handler

### ISSUE THREAD

**Issue by alltheseas** [alltheseas/nostrability#7]
Damus 🚫

**Comment by alltheseas** [comment]
See also https://github.com/nostrability/nostrability/issues/6

**Comment by fabianfabian** [comment]
Nostur supports NIP-89 but only lists recommended web apps, due to a limitation on iOS that restricts finding out if an app is installed before deep linking

**Comment by alltheseas** [comment]
added client tag column

**Comment by alltheseas** [comment]
![image](https://github.com/user-attachments/assets/7fb3b41a-21d0-4b2c-a059-e2d84a665d9f) 

Nostur client post metadata setting

**Comment by alltheseas** [comment]
added snort thanks to @karnagebitcoin testing

cc @v0l

**Comment by v0l** [comment]
same in Snort, doesnt (yet) sort by recommendations

**Comment by alltheseas** [comment]
coracle example https://video.nostr.build/6b9c22f94ccf9d8bbc7c7873bf7caa1e1636ff97ed3fa4fea7f2bab2d9e6eae6.mp4


https://github.com/user-attachments/assets/038e5668-4170-46c4-9c38-b45ca820c410

**Comment by alltheseas** [comment]
Added link to nip-89 talk by @pablof7z via nostrasia

**Comment by alltheseas** [comment]
jumble social

![Image](https://github.com/user-attachments/assets/ca7abdea-bed1-4f0b-aa30-a7dc201bc453)

 utilizes client tag as an opt-in advanced feature

**Comment by alltheseas** [comment]
ditto.pub client tags 

![Image](https://github.com/user-attachments/assets/9704cf0c-1026-4f4e-9330-5f0b79380d8f)

**Comment by alexgleason** [comment]
I opened an MR to add client tags to Primal Web: https://github.com/PrimalHQ/primal-web-app/pull/126

**Comment by erskingardner** [comment]
@vitorpamplona - any chance you want to add similar to amethyst?

**Comment by alltheseas** [comment]
@vitorpamplona could you place client tags in the note JSON - i.e. do not make visible similar to how to tags are not made visible in the UI?

Regarding the latter, could you make client tags opt-in, and suggest there might be unknown risks of advertising which app you use? 

The benefit of including tags will be decent for interoperability purposes. I imagine for a DM app you can quickly determine (in)compatibility of your counterparty. 

For trouble between apps we won't need to go on a Sherlock Holmes adventure to figure out which client requires a interop or bugfix.

**Comment by vitorpamplona** [comment]
> Regarding the latter, could you make client tags opt-in

If I make it opt-in, then less than 1% of users will activate it. Most of our users have never changed any settings at all. It doesn't seem to do much for interoperability. 

> The benefit of including tags will be decent for interoperability purposes.

I am not sure whether this helps or not. If it works and using the same clients is important for interoperability, it centralizes on the more popular clients. If it doesn't, it's just noise for users and minimally reliable analytics for us. I don't think it makes sense to let users do things (say, send a NIP-17 DM) based on which client tags their friends use. Besides, NIP-17 already has a mechanism to advertise usage. 

Overflowing the client tag to determine compatibility doesn't make much sense to me. Especially not without client versions being properly documented in each event.

**Comment by alltheseas** [comment]
> Especially not without client versions being properly documented in each event.

For troubleshooting broken experiences, of which there are many, this sound great. How might we add this to client tags?

**Comment by alltheseas** [comment]
> If I make it opt-in, then less than 1% of users will activate it

This is great, and representative. 1% of nostr users (or less) are the power users who continuously report bugs. You know who these peopl are for amethyst, and other devs know them for their app.

> it centralizes on the more popular clients. If it doesn't, it's just noise for users and minimally reliable analytics for 

Right now (without client tags) no one knows which app(s) work with each other app(s). The effect of this is centralizing to the most popular 2-3 apps. Whereas, client tags serve as a discovery mechanism, and info-labels for other apps.

**Comment by vitorpamplona** [comment]
> For troubleshooting broken experiences, of which there are many, this sound great. How might we add this to e.g. client tags and/or elsewhere?

Again, I don't think this makes any sense for compatibility. We cannot keep track of what each client is dong in each version to know if our users can do or not do certain things.

> Right now (without client tags) no one knows which app(s) work with each other app(s). 

I don't think client tags solve for this at all. Even with client versions in them. Making each client know each other client version so that it can provide appropriate guidance to the user is crazy. 

> The effect of this is centralizing to the most popular 2-3 apps. Whereas, client tags serve as a discovery mechanism, and info-labels for other apps.

This is true, but only to discover other apps dealing with the same event kinds as the current app already displays correctly. 

It doesn't help with the NIP-17 example because either (i) the user is using a NIP-17 client and it can decrypt and see the messages or (ii) the user is not using a NIP-17 client and it will never display the client tags from those events because the client doesn't not even download the events themselves.

**Comment by alltheseas** [comment]
@auggie-lahey what do you think about client tags and nostrCI mission?

**Comment by alexgleason** [comment]
@vitorpamplona The reason you should do this is because it's fun. And it's easy, which is a rare combination.

I'm clicking every post now to see if there's a client tag. It makes people feel competitive and provokes discussion. I think it's actually very valuable information. Not to mention free advertising for Amethyst. I think users want it and you should want it. Plus it opens doors to features like: zap splits to clients, client-only feeds, trending clients, etc etc. It's like win-win-win. I also didn't see the value initially until I added it and started clicking around. It's a lot of bang for the buck IMO.

**Comment by vitorpamplona** [comment]
Yeah, fun is the only reason to add it. But I think that fun is just for us, not for users.

**Comment by alltheseas** [comment]
> Yeah, fun is the only reason to add it. But I think that fun is just for us, not for users.

What do you think about `client tag` @Silberengel in context of nostr app discovery?

**Comment by Silberengel** [comment]
I'm fine with it. I turn it on.
